Cannot build HDF5 @1.10.0-patch1 +mpi

I am trying to build HDF5 on a Linux workstation with the `+mpi` option. This build fails in the configure stage with this error:
```
configure:4528: checking whether the C compiler works
configure:4550: /xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/bin/mpicc       conftest.c  >&5
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_proc_is_bound'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_rml_send_callback'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_grpcomm_base'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_rml_recv_callback'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_state_base_framework'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_rml_recv_cb_t_class'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_proc_state_to_str'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_grpcomm_collective_t_class'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_rmaps_base_set_mapping_policy'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_state'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_info_close_components'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_rmaps_base_set_ranking_policy'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_info_show_orte_version'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_grpcomm_base_store_modex'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_plm_base_fork_hnp'
/xfs1/eschnetter/src/spack/opt/spack/linux-x86_64/gcc-6.1.0-spack/openmpi-1.10.2-thgejjepeloy5b3wgat2ax6t2d47poon/lib/libmpi.so: undefined reference to `orte_info_register_framework_params'
collect2: error: ld returned 1 exit status
configure:4554: $? = 1
configure:4592: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "HDF5"
| #define PACKAGE_TARNAME "hdf5"
| #define PACKAGE_VERSION "1.10.0"
| #define PACKAGE_STRING "HDF5 1.10.0"
| #define PACKAGE_BUGREPORT "help@hdfgroup.org"
| #define PACKAGE_URL ""
| #define PACKAGE "hdf5"
| #define VERSION "1.10.0"
| /* end confdefs.h.  */
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:4597: error: in `/tmp/eschnetter/spack-stage/spack-stage-XI7f3H/hdf5-1.10.0':
configure:4599: error: C compiler cannot create executables
```

This used to work a few weeks ago. Any pointers would be appreciated.

--- a/var/spack/repos/builtin/packages/openmpi/package.py
+++ b/var/spack/repos/builtin/packages/openmpi/package.py
@@ -142,7 +142,7 @@ class Openmpi(Package):
         ])
         if '+verbs' in spec:
             path = _verbs_dir()
-            if path is not None:
+            if path is not None and path not in ('/usr', '/usr/local'):
                 config_args.append('--with-%s=%s' % (self.verbs, path))
             else:
                 config_args.append('--with-%s' % self.verbs)

